File: day81-90/day85/1277. Count Square Submatrices with All Ones.py
from typing import List

class Solution:
    def countSquares(self, matrix: List[List[int]]) -> int:
        m, n = len(matrix), len(matrix[0])
        total_squares = 0
        
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 1:
                    if i > 0 and j > 0:
                        matrix[i][j] = min(matrix[i-1][j], matrix[i][j-1], matrix[i-1][j-1]) + 1
                    total_squares += matrix[i][j]
        
        return total_squares
    
sol=Solution()
matrix = [
  [1,0,1],
  [1,1,0],
  [1,1,0]
]
print(sol.countSquares(matrix))

File: day81-90/day85/1277. Count Square Submatrices with All Ones.ts
function countSquares(matrix: number[][]): number {
  const m: number = matrix.length;
  const n: number = matrix[0].length;
  let totalSquares: number = 0;

  for (let i: number = 0; i < m; i++) {
    for (let j: number = 0; j < n; j++) {
      if (matrix[i][j] == 1) {
        if (i > 0 && j > 0) {
          matrix[i][j] =
            Math.min(matrix[i - 1][j], matrix[i][j - 1], matrix[i - 1][j - 1]) +
            1;
        }
        totalSquares += matrix[i][j];
      }
    }
  }

  return totalSquares;
}
